.github/workflows/cni-validator.yml

```diff
@@ -0,0 +1,114 @@
+name: CNI Validator
```

Comment:
We probably don't need a separate workflow for this. Many of these checks are redundant with those in the policy controller workflow. Instead, we probably want to extract a common _rust_ workflow and leave the policy integration tests in the policy workflow.

Reply:
Yup, you're completely right. I noticed that after failing CI a few times. I'll focus on that next.